File: src/main/java/com/jozufozu/exnihiloomnia/common/registries/ingredients/IngredientNBT.java
package com.jozufozu.exnihiloomnia.common.registries.ingredients;

import net.minecraft.item.ItemStack;
import net.minecraft.item.crafting.Ingredient;

import javax.annotation.Nullable;

/**
 * Forge's has a protected constructor
 */
public class IngredientNBT extends Ingredient
{
    private final ItemStack stack;
    public IngredientNBT(ItemStack stack)
    {
        super(stack);
        this.stack = stack;
    }
    
    @Override
    public boolean apply(@Nullable ItemStack input)
    {
        if (input == null)
            return false;
        return ItemStack.areItemStacksEqualUsingNBTShareTag(this.stack, input);
    }
}

File: src/main/kotlin/com/jozufozu/exnihiloomnia/common/registries/ingredients/IngredientNBT.kt
package com.jozufozu.exnihiloomnia.common.registries.ingredients

import net.minecraft.item.ItemStack
import net.minecraft.item.crafting.Ingredient

/**
 * Forge's has a protected constructor
 */
class IngredientNBT(private val stack: ItemStack) : Ingredient(stack) {

    override fun apply(input: ItemStack?): Boolean {
        return if (input == null) false else ItemStack.areItemStacksEqualUsingNBTShareTag(this.stack, input)
    }
}
